### PR TITLE
move tests for bugs to regression module and set readable test names

### PR DESCRIPTION
**Pull request by miherlosev**
/cc @inikulin @LavrovArtem

### CODE REVIEW

**Comment by miherlosev** [comment]
FRP

**Comment by inikulin** [comment]
In general looks good with some minor style nits. BTW check tests name spelling with @VasilyStrelyaev

**Comment by inikulin** [comment]
\r-

File: test/client/fixtures/sandboxes/dom/dom-accessor-wrappers/setters-test.js [hunk 1]
```diff
@@ -281,34 +301,7 @@ asyncTest('body.innerHtml in iframe (with a.href)', function () {
...
-        value:   ''
-    };
-
-    strictEqual(eval(processScript('obj.value')), '');
-    /* eslint-enable no-unused-vars */
-});
-
-//T228218: Page script can\'t set the href attribute with \'malito\' to an element in an iFrame
-test('href attribute with \'malito\' value to an iframe element', function () {
+test('setting the link.href attribute to \'malito\' in iframe (T228218)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/dom/dom-accessor-wrappers/setters-test.js]: mailto

File: test/client/fixtures/sandboxes/dom/document-test.js [hunk 2]
```diff
@@ -190,9 +122,51 @@ if (!Browser.isMozilla) {
...
+    strictEqual(scripts[0].text.replace(/\s*/g, ''), processedScript);
+
+    $iframe.remove();
+    $div.remove();
+});
+
 if (Browser.isMozilla || Browser.isIE11) {
-    //T239109 - TD 15.1 - Hummerhead script error after perform search on http://livejournal.com page
-    asyncTest('override window methods after document.write', function () {
+    asyncTest('override window methods after document.write calling (T239109)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/dom/document-test.js]: calling -> call

File: test/client/fixtures/sandboxes/dom/document-test.js [hunk 3]
```diff
@@ -227,8 +201,31 @@ if (Browser.isMozilla || Browser.isIE11) {
...
+if (!Browser.isMozilla) {
+    asyncTest('document.write([]) in iframe (T239131)', function () {
+        var iframe = document.createElement('iframe');
+
+        iframe.id = 'test04';
+
+        var loadHandler = function () {
+            iframe.removeEventListener('load', loadHandler);
+
+            // Some browsers remove they documentElement after "write([])" calling.
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/dom/document-test.js]: 1) they? (you must've meant 'their' or 'the')
2) calling -> call

File: test/client/fixtures/sandboxes/event/event-simulator-test.js [hunk 4]
```diff
@@ -288,7 +289,7 @@ if (Browser.isIE) {
         });
     }
 
-    test('B237405 - window.event contains toElement and fromElement properties for mouseout and mouseover events', function () {
+    test('window.event must contains toElement and fromElement properties for mouseout and mouseover events (B237405)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/event/event-simulator-test.js]: must contain

File: test/client/fixtures/sandboxes/event/event-test.js [hunk 5]
```diff
@@ -154,8 +165,7 @@ test('document.addEventListener', function () {
     strictEqual(docClickRaisedCount, 1);
 });
 
-//Q532574 - Fire and dispatch events from code, events created by different ways
-test('combination of attachEvent and addEventListener', function () {
+test('firing and dispatching the events created by different ways (Q532574)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/event/event-test.js]: in different ways

File: test/client/fixtures/sandboxes/event/internal-listeners-test.js [hunk 6]
```diff
@@ -345,8 +345,9 @@ $(document).ready(function () {
         Listeners.removeInternalEventListener(container, [event], cancelHandlers);
     });
 
-    //T233158 - Wrong test run for mouse click in IE
-    test('special case for mourse click in IE(document handlers)', function () {
+    module('regression');
+
+    test('only one from several handlers must be called (document handlers) (T233158)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/event/internal-listeners-test.js]: one of several

File: test/client/fixtures/sandboxes/event/internal-listeners-test.js [hunk 7]
```diff
@@ -407,8 +408,7 @@ $(document).ready(function () {
         document.removeEventListener(event, clickHandler, false);
     });
 
-    //T233158 - Wrong test run for mouse click in IE
-    test('special case for mourse click in IE (body handlers)', function () {
+    test('only one from several handlers must be called (body handlers) (T233158)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/event/internal-listeners-test.js]: one of several

File: test/client/fixtures/sandboxes/event/internal-listeners-test.js [hunk 8]
```diff
@@ -441,8 +441,7 @@ $(document).ready(function () {
         $body.off('click', clickHandler);
     });
 
-    //T233158 - Wrong test run for mouse click in IE
-    test('special case for mourse click in IE (element handlers)', function () {
+    test('only one from several handlers must be called (element handlers) (T233158)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/event/internal-listeners-test.js]: one of several

File: test/client/fixtures/sandboxes/event/internal-listeners-test.js [hunk 9]
```diff
@@ -473,8 +472,7 @@ $(document).ready(function () {
     });
 
     if (Browser.isIE && Browser.version >= 10) {
-        //T233158 - Wrong test run for mouse click in IE (document pointer event handlers)
-        test('MSPointerDown, pointerdown', function () {
+        test('only one from several handlers must be called (MSPointerDown, pointerdown combination) (T233158)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/event/internal-listeners-test.js]: one of several

File: test/client/fixtures/sandboxes/shadow-ui-test.js [hunk 10]
```diff
@@ -479,3 +408,72 @@ test('querySelectorAll', function () {
     strictEqual(elems.length, 1);
     strictEqual(elems[0].id, 'pageElem');
 });
+
+module('regression');
+
+asyncTest('after clean up iframe.body.innerHtml ShadowUI\'s root must be exist(T225944)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/shadow-ui-test.js]: must exist

File: test/client/fixtures/sandboxes/shadow-ui-test.js [hunk 11]
```diff
@@ -479,3 +408,72 @@ test('querySelectorAll', function () {
...
+        strictEqual($root.parent().parent().parent()[0], this.contentDocument);
+
+        $iframe.remove();
+        start();
+    });
+
+    $iframe.appendTo('body');
+});
+
+test('getElementsByClassName, querySelectorAll must not overriden (T195358)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/shadow-ui-test.js]: must not be overriden

File: test/client/fixtures/sandboxes/shadow-ui-test.js [hunk 12]
```diff
@@ -479,3 +408,72 @@ test('querySelectorAll', function () {
...
+    var doc             = {};
+    var nativeMethRegEx = /^[^{]+\{\s*\[native \w/;
+
+    ShadowUI.init(null, doc);
+
+    ok(nativeMethRegEx.test(doc.getElementsByClassName));
+    ok(nativeMethRegEx.test(doc.querySelectorAll));
+});
+
+test('ShadowUI\'s root must be last children after adding a new element (T239689)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/shadow-ui-test.js]: must be the last child? or something else?

File: test/client/fixtures/sandboxes/xhr-test.js [hunk 13]
```diff
@@ -1,5 +1,6 @@
-//T239198: TD15.1 - web page doesnt display correctly (hellomonday.com)
-asyncTest('correct pass paremeters into xhr event handlers', function () {
+module('regression');
+
+asyncTest('parameters must be pass correctly in xhr event handlers (T239198)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/xhr-test.js]: must pass

File: test/client/fixtures/util/html-test.js [hunk 14]
```diff
@@ -367,3 +325,45 @@ asyncTest('real big page', function () {
     }).appendTo('body');
 });
 
+
+module('regression');
+
+test('script.innerHtml must be cleanup (T226885)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/util/html-test.js]: must be cleaned up ?

File: test/client/fixtures/util/html-test.js [hunk 15]
```diff
@@ -367,3 +325,45 @@ asyncTest('real big page', function () {
...
+    var code    = 'var t = 1;';
+    var $script = $('<script>' + code + '<\/script>');
+
+    notEqual($script[0].innerHTML.replace(/^\s*|\s*$/g, ''), code);
+    strictEqual(eval(processScript('$script[0].innerHTML')).replace(/^\s*|\s*$/g, ''), code);
+
+    $script.remove();
+});
+
+test('markup with special characters must be cleanup correctly (T112153)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/util/html-test.js]: must be cleaned up ?

File: test/client/fixtures/util/html-test.js [hunk 16]
```diff
@@ -367,3 +325,45 @@ asyncTest('real big page', function () {
...
+            'data-actionsheet-context="#:ID#">Reply</a>\n' +
+            '<h2>#: Subject#</h2>\n' +
+            '<p>#: Text#</p>';
+
+    // </script\>
+
+    strictEqual(Html.cleanUpHtml(html).replace(/\s/g, ''), html.replace(/\s/g, ''));
+});
+
+test('html and body attributes must be process (T226655)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/util/html-test.js]: must be processed

**Comment by miherlosev** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by inikulin** [comment]
lgtm

**Pull request merged**